### PR TITLE
impute NAs in dcalib autoplot

### PR DESCRIPTION
**Pull request by RaphaelS1**

### CODE REVIEW

File: R/autoplot.R [hunk 1]
```diff
@@ -234,8 +236,13 @@ autoplot.PredictionSurv = function(object, type = "dcalib",
 
     "dcalib" = {
       p = seq.int(0, 1, length.out = cuts)
-      q = map_dbl(p, function(.x) {
-        sum(object$truth[, 1L] <= as.numeric(object$distr$quantile(.x)), na.rm = TRUE) / length(object$row_ids)
+      true_times = object$truth[, 1L]
+      q = mlr3misc::map_dbl(p, function(.x) {
```
Comment (RaphaelS1) [inline R/autoplot.R:240]:
```suggestion
      q = map_dbl(p, function(.x) {
```

**Comment by bblodfon** [comment]
ha! I did some things with the `autoplot` on the Breslow PR, should bring them here

**Comment by bblodfon** [comment]
Nice @RaphaelS1! Some comments (I could help changing these):
- Why we check the generated plots (as `svg`) in the tests? Looks like a test that is destined to break as it happens with the latest ubuntu version. I would suggest removing the `vdiffr::expect_doppelganger(...)`? (which has already been done to some)
- `row_ids` parameter for type = `calib` should be the train set rows, right? I ran the example to check the `calib` plot of `surv.kaplan` and it's not the exact same curve. Doc needs to be updated as well
- Calibration plot in the literature is commonly referred the `dcalib` type you implemented here. I would vote to have that as default option and maybe briefly mention that.
- I had noticed that `dcalib` plots looked not complete so now `extend_quantiles = TRUE` fixes that. Maybe have that as default make more sense?

**Comment by RaphaelS1** [comment]
> Why we check the generated plots (as svg) in the tests? Looks like a test that is destined to break as it happens with the latest ubuntu version. I would suggest removing the vdiffr::expect_doppelganger(...)? (which has already been done to some)

This is an mlr3 wide decision

> row_ids parameter for type = calib should be the train set rows, right? (test set doesn't affect the survival predictions). I executed the example to check the calib plot of surv.kaplan and it's not the exact same curve (ie KM vs KM). Doc would need to be updated as well.

No, you compare to the Kaplan-Meier on the testing set  to see if it's well calibrated as you're comparing predictions with the truth

> Calibration plot in the literature is commonly referred the dcalib type you implemented here. I would vote to have that as default option and maybe briefly mention that.

Disagree. dcalib is *very* new, whereas graphical comparisons to a baseline are much more established

> I had noticed that dcalib plots looked not complete so now extend_quantiles = TRUE fixes that. Maybe have that as default make more sense?

Disagree. We deliberately removed the automatic extension of survival probabilities (so S(Inf) = 1 and S(0) = 1) because of the c-hacking paper findings, by having that as the default you're basically doing the same thing. It's good enough for the plot but it's technically incorrect

**Comment by bblodfon** [comment]
> No, you compare to the Kaplan-Meier on the testing set to see if it's well calibrated as you're comparing predictions with the truth

Yes! What I am trying to understand is this: in the [code](https://github.com/mlr-org/mlr3proba/blob/main/R/autoplot.R#L221) we use the `rows_ids` both to define the train and the test set of the KM (which should be different I guess? and the test set doesn't really matter as in the [predict](https://github.com/mlr-org/mlr3proba/blob/main/R/LearnerSurvKaplan.R#L41) method we just replicate the survival probabilities from the train set and these are averaged in the ploting function). Then when we try to assess the calibration of KM against itself we don't get an exact match (ie it should be `Pred` == `KM` below, right?):
``` r
library(mlr3)
library(mlr3proba)
library(mlr3viz)

learn = lrn("surv.kaplan")
task = tsk("unemployment")
p = learn$train(task, row_ids = 1:300)$predict(task, row_ids = 301:400)
autoplot(p, type = "calib", task = task, row_ids = 301:400)
```



``` r
autoplot(p, type = "calib", task = task, row_ids = 1:300) # use train set
```



<sup>Created on 2023-12-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by RaphaelS1** [comment]
@bblodfon I'm not sure I understand your point... 1) as you said it doesn't matter what's used for train/test in KM and it should be the same as we're estimating the empirical survival function, i.e. technically it's an unsupervised method; 2) your plots are testing two completely separate things, in the first you're testing if the model is well-calibrated to the test data (i.e., an out-of-sample evaluation metric) and in the second you're testing if the model is fit well to the training data.

**Comment by bblodfon** [comment]
> in the first you're testing if the model is well-calibrated to the test data (i.e., an out-of-sample evaluation metric)

Yes, and since the **model itself** is the KM, the curves should match as calibration tests against predictions from KM, right? ie the first plot should look like the second, that's what I think (but I may have the wrong understanding here!)

**Comment by bblodfon** [comment]
KM is not used as a benchmark model here, only as the (unsupervised) estimator of the true survival function in the test set, so implementation is correct
